### PR TITLE
Implement VOF phase fraction filter

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
# Description of the problem

- Sharpening tends to pull the fluid together (similar to the effect of artificial surface tension forces)

# Description of the solution

- Implement a phase fraction filter. At the moment, there is 2 filter options ("none" and "tanh").
- Both filtered and unfiltered solutions are outputted.

# How Has This Been Tested?

- Visual verification with the Rayleigh-Taylor example
- Application test (applications_tests/gls_navier_stokes_2d/gls_vof_tanh_filter)

# Documentation

- [x] doc/source/parameters/cfd/volume_of_fluid.rst

# Future changes

- Use the filtered phase fraction when calculating the physical properties.

### CODE REVIEW

**Review by blaisb — APPROVED** [review]
No comments, this is an amazingly well done PR :)!
The only thing missing would be to add a small application test where this is used (or add this to one of the application test) :)!

**Comment by AmishgaAlphonius** [comment]
> No comments, this is an amazingly well done PR :)!
> The only thing missing would be to add a small application test where this is used (or add this to one of the application test) :)!

Thank you!
You're right! l'll add that tomorrow! :)

**Comment by blaisb** [comment]
I'll wait for @hepap review then i'll merge :)!

File: source/core/parameters_multiphysics.cc [hunk 1]
```diff
@@ -507,3 +509,66 @@ Parameters::VOF_SurfaceTensionForce::parse_parameters(ParameterHandler &prm)
...
+      "10",
+      Patterns::Double(),
+      "This parameter appears in the tanh filter function. It influence "
+      "the thickness and the shape of the interface. For higher values of "
+      "beta, a thinner and 'sharper/pixelated' interface will be seen.");
+    prm.declare_entry("verbosity",
+                      "quiet",
+                      Patterns::Selection("quiet|verbose|extra verbose"),
+                      "States whether the filtered data should be printed "
+                      "Choices are <quiet|verbose|extra verbose>.");
```
Comment (hepap) [inline source/core/parameters_multiphysics.cc:539]:
```suggestion
                      "Choices are <quiet|verbose>.");
```

File: source/core/parameters_multiphysics.cc [hunk 2]
```diff
@@ -507,3 +509,66 @@ Parameters::VOF_SurfaceTensionForce::parse_parameters(ParameterHandler &prm)
...
+    beta = prm.get_double("beta");
+
+    // Verbosity
+    const std::string filter_v = prm.get("verbosity");
+    if (filter_v == "verbose")
+      verbosity = Parameters::Verbosity::verbose;
+    else if (filter_v == "quiet")
+      verbosity = Parameters::Verbosity::quiet;
+    else if (filter_v == "extra verbose")
+      verbosity = Parameters::Verbosity::extra_verbose;
```
Comment (hepap) [inline source/core/parameters_multiphysics.cc:569]:
```suggestion
```

File: include/solvers/vof_filter.h [hunk 3]
```diff
@@ -0,0 +1,100 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2020 by the Lethe authors
```
Comment (hepap) [inline include/solvers/vof_filter.h:3]:
```suggestion
 * Copyright (C) 2019 - by the Lethe authors
```

**Review by hepap — APPROVED** [review]
Nice PR :) Just a small thing about the verbosity in the parameter file!

**Pull request merged**